source/backend/api/Services/AcquisitionFileService.cs

```diff
@@ -393,6 +409,31 @@ public IEnumerable<PimsAgreement> UpdateAgreements(long acquisitionFileId, List<
         {
             _user.ThrowInvalidAccessToAcquisitionFile(_userRepository, _acqFileRepository, acquisitionFileId);
 
+            var currentAcquisitionStatus = GetCurrentAcquisitionStatus(acquisitionFileId);
+
+            var currentAgreements = _agreementRepository.GetAgreementsByAcquisitionFile(acquisitionFileId);
+
+            var toBeUpdated = currentAgreements.Where(ca => agreements.Any(na => ca.AgreementId == na.AgreementId && !ca.IsEqual(na)));
+            var toBeDeleted = currentAgreements.Where(ca => !agreements.Any(na => ca.AgreementId == na.AgreementId));
+
+            foreach (var agreement in toBeUpdated)
```

Comment:
is there a reason that agreements being edited and agreements being deleted are being split out into their own loops? Seems like the business logic that is being executed is identical, sort of adds an implication that updates should not be handles the same as deletes.

Reply:
Although true, doing the loop on each would be the same as doing the loop on an aggregated collection.